core/src/edu/asu/c3simulator/screens/TaskManagement.java

```diff
@@ -0,0 +1,273 @@
+		
+		leftSection.add(tasksListLabel).fillX().align(Align.center);
+		leftSection.row();
+		leftSection.add(chosenTasksScroll).expand().fill().center();
+		leftSection.row();
+		leftSection.add(removeTasksButton).fillX().center();
+		
+		this.add(leftSection).expand().fill();
+		this.add(rightSection).expand().fill();
+		this.setPosition(stage.getWidth() / 2 - this.getWidth() / 2, stage.getHeight()
+				/ 2 - this.getHeight() / 2);
+		
+		updateAvailableTasks();
+		updateChosenTasks();
+	}
+	
+	/**
+	 * This method is called to update the list of available tasks.
+	 */
+	public void updateAvailableTasks()
+	{
+		ArrayList<Task> availableTasks = getAvailableTasks();
+		actionBasedTasksList.clear();
+		nonActionBasedTasksList.clear();
+		
+		for (Task task : availableTasks)
+		{
+			CheckBox newTaskCheckBox = new CheckBox(" " + task.getName(), skin);
+			displayedTasks.put(newTaskCheckBox, task);
+			if (task.getType().equalsIgnoreCase("Action Based"))
+			{
+				actionBasedTasksList.addActor(newTaskCheckBox);
+				actionBasedTasksGroup.add(newTaskCheckBox);
+			}
+			else
+			{
+				nonActionBasedTasksList.addActor(newTaskCheckBox);
+				nonActionBasedTasksGroup.add(newTaskCheckBox);
+			}
+		}
+		
+		actionBasedTasksList.left();
+		nonActionBasedTasksList.left();
+	}
+	
+	/**
+	 * This method creates a testing shell of different tasks to test and demo the GUI.
+	 * 
+	 * @return an ArrayList of the available tasks that are going to be displayed.
+	 */
+	private ArrayList<Task> getAvailableTasks()
+	{
+		String[] actionBasedList = { "task 1.1", "task 2.1", "task 3.1", "task 4.1",
+				"task 5.1", "task 6.1", "task 7.1", "task 8.1", "task 9.1", "task 10.1" };
+		String[] nonActionBasedList = { "task 1.2", "task 2.2", "task 3.2", "task 4.2",
+				"task 5.2", "task 6.2", "task 7.2", "task 8.2", "task 9.2", "task 10.2" };
+		
+		ArrayList<Task> availableTasks = new ArrayList<Task>();
+		
+		for (String taskName : actionBasedList)
+			availableTasks.add(new Task(taskName, "Action Based"));
+		
+		for (String taskName : nonActionBasedList)
+			availableTasks.add(new Task(taskName, "Non-Action Based"));
+		
+		return availableTasks;
+	}
+	
+	/**
+	 * This method deals with obtaining the currently chosen tasks by the user from the
+	 * data layer and display it at the beginning.
+	 */
+	public void updateChosenTasks()
+	{
+		// TODO get current chosen tasks
+	}
+	
+	/**
+	 * This method gets the current list in the chosen tasks column and return it in array
+	 * form.
+	 * 
+	 * @return an ArrayList of the chosen tasks.
+	 */
+	public ArrayList<Task> getChosenTasks()
+	{
+		ArrayList<Task> chosenTasks = new ArrayList<Task>();
+		
+		Array<Button> chosenTaskButtons = chosenTasksGroup.getButtons();
+		for (Button checkedButton : chosenTaskButtons)
+		{
+			CheckBox checkBoxButton = (CheckBox) checkedButton;
+			chosenTasks.add(displayedTasks.get(checkBoxButton));
+		}
+		
+		return chosenTasks;
+	}
```

Comment:
Flag for unit testing

GUI cases do not need to be unit tested - we'll verify those in other ways.
However, I'd write some unit tests to verify the following:
- null is not returned
- an empty list is returned if called directly after instantiation

It looks like this result will be achieved by this code, but unit tests will keep this verified in future updates